lib/model/app_model/list_invoices.dart

```diff
@@ -9,80 +10,60 @@ class AppListInvoices {
   factory AppListInvoices.fromJSON(Map<String, dynamic> json,
       {bool snackCase = false, bool isObject = false, String? status}) {
     var invoices = json.withKey("invoices", snackCase: snackCase) as List;
+    List<AppInvoice> list = [];
+
     if (invoices.isNotEmpty) {
-      var appInvoices = invoices.map((invoice) {
-        var temp = AppInvoice.fromJSON(invoice,
+      for (var invoices in invoices) {
+        AppInvoice temp = AppInvoice.fromJSON(invoices,
             snackCase: snackCase, isObject: isObject);
-        if (temp.status == status) {
-          return temp;
+
+        /// Filtering the invoices on the basis of status
+        if (temp.status == "paid") {
+          list.add(temp);
         }
-      }).toList();
-      if (appInvoices.contains(null)) {
-        return AppListInvoices();
       }
-      return AppListInvoices(invoice: appInvoices as List<AppInvoice>);
+      return AppListInvoices(invoice: list);
     } else {
       return AppListInvoices();
     }
   }
 }
 
 class AppInvoice {
-  /// payment hash
-  final String paymentHash;
-
   /// bolt11 identifier
   final String bolt11;
 
   /// The quantity of Bitcoin in millisatoshi
   final String amount;
 
-  /// If the invoice is paid or not on the blockchain
-  final String status;
-
-  /// The Invoice paid time
-  final String paidTime;
+  final String identifier;
 
-  /// The Invoice description
-  final String label;
+  final String status;
 
-  /// The Invoice description
   final String description;
 
-  /// flag to identify invoice
-  final String identifier;
-
   AppInvoice(
-      {required this.paymentHash,
+      {required this.bolt11,
       required this.amount,
       required this.status,
-      required this.paidTime,
-      required this.bolt11,
       required this.description,
-      required this.label,
       this.identifier = "invoice"});
 
   factory AppInvoice.fromJSON(Map<String, dynamic> json,
       {bool snackCase = false, bool isObject = false}) {
     LogManager.getInstance.debug("$json");
-    var bolt11 = json.withKey("bolt11", snackCase: snackCase) ??
-        json.withKey("bolt12", snackCase: snackCase);
-    var paymentHash = json.withKey("paymentHash", snackCase: snackCase);
+    var bolt11 = json.withKey("bolt11", snackCase: snackCase);
     var status = json.withKey("status", snackCase: snackCase);
-    var received = json.parseMsat(
-        key: "amountReceivedMsat", snackCase: snackCase, isObject: isObject);
-    received.toString();
-    var paidAt = json.withKey("paidAt", snackCase: snackCase);
     var description = json.withKey("description", snackCase: snackCase);
-    var label = json.withKey("label", snackCase: snackCase);
+    pay_req.Bolt11PaymentRequest req = pay_req.Bolt11PaymentRequest(bolt11);
+    LogManager.getInstance.debug(
+        "NEW RESPONSE HERE : ${req.amount} ${req.paymentRequest} ${req.prefix} ${req.signature} ${req.tags} ${req.timestamp}");
+    var amount = req.amount;
```

Comment:
possible that this is null? from my that I post before?

Reply:
I guess the bolt12 invoices are not handeled and the post you shared above has an bolt12 invoice.

Reply:
well we should not crash, but just not decode it and spring the raw bolt12